block/mempool.go

```diff
@@ -164,9 +165,18 @@ func (m *LanedMempool) Contains(tx sdk.Tx) (contains bool) {
 }
 
 // Registry returns the mempool's lane registry.
-func (m *LanedMempool) Registry(ctx sdk.Context) (newRegistry []Lane, err error) {
-	if m.moduleLaneFetcher == nil {
-		return m.registry, fmt.Errorf("module lane fetcher not set")
+//
+// CONTRACT: this function panics if it fails.  It should be wrapped in a recover() mechanism.
+// This function will NEVER panic if x/blocksdk is disabled.
+func (m *LanedMempool) Registry(ctx sdk.Context) []Lane {
```

Comment:
Why panic instead of err here?

Reply:
Adding the `err` was the API breaking change that we made when x/blocksdk was added.  But to backport, we need to retain the existing API, so it will only panic when the module is enabled

Reply:
kek

Reply:
gud semantic versioning